common/persistence/visibility/visibility_manager_impl.go

```diff
@@ -140,6 +144,81 @@ func (p *visibilityManagerImpl) ListWorkflowExecutions(
 	return p.convertInternalListResponse(response)
 }
 
+func (p *visibilityManagerImpl) ListChasmExecutions(
+	ctx context.Context,
+	request *manager.ListChasmExecutionsRequest,
+) (*manager.ListChasmExecutionsResponse, error) {
+	response, err := p.store.ListChasmExecutions(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	executions := make([]*manager.ChasmExecutionInfo, 0, len(response.Executions))
+	for _, exec := range response.Executions {
+		combinedMemo, err := deserializeMemo(exec.Memo)
+		if err != nil {
+			return nil, err
+		}
+
+		var userMemo *commonpb.Memo
+		var chasmMemoPayload *commonpb.Payload
+
+		// Check if archetype matches to decide how to split memo
```

Comment:
hmm why not just check if `__chasm__` is a key in the `combinedMemo`? 

if we are worrying about potential conflict (though I feel that chance is very low...), I'd probably just check if TemporalNamespaceDivision string is a number or not (just the first char should be enough), since we are going to use this method to query across archetypes, e.g. in namespace migration case, I just want all visibility records.